packages/nimble-components/src/table/specs/table-column-specs/table-column-file-size.md

```diff
@@ -0,0 +1,294 @@
+# File Size Table Column
+
+## Overview
+
+The file size table column will provide a way to visualize numeric data that represents a file size (or other binary data size). It will automatically convert the raw value to use an appropriate display unit.
+
+### Background
+
+[Work Item](https://ni.visualstudio.com/DevCentral/_workitems/edit/2526114)
+
+### Non-goals
+
+-   Configurable number of decimal places
+-   Configurable units (e.g. display all values in KB)
+-   Grouping by value ranges/buckets
+
+### Features
+
+-   Converts the byte value to largest unit (e.g. KB, MB, GB) that results in a value of >=1
+-   Option to use 1024-based units (e.g. KiB) or 1000-based units (e.g. KB)
+-   Displays up to one decimal place
+-   Sorts/groups by numeric value (number of bytes) rather than display string
+-   Localized number and unit strings
+
+### Risks and Challenges
+
+### Prior Art/Examples
+
+Size column in [SLE Files grid](https://dev.lifecyclesolutions.ni.com/files/)
+
+Windows explorer (details view)
+
+---
+
+## Design
+
+Below is an example of how the `nimble-table-column-file-size` would be used within a `nimble-table`:
+
+```HTML
+<nimble-table>
+    <nimble-table-column-file-size field-name="size">Size</nimble-table-column-file-size>
+</nimble-table>
+```
+
+### API
+
+`nimble-table-column-file-size`
+
+-   `field-name` - name of the record field containing the data. Data must be a `number` byte count.
+-   `unit-type` - "binary" (KiB, MiB, GiB, TiB, PiB) or "decimal" (KB, MB, GB, TB, PB); defaults to "decimal"
+
+The component will extend `TableColumnTextBase`, thereby including the APIs for a **groupable** and **fractional-width** column.
+
+### Anatomy
+
+### Angular integration
+
+An Angular directive will be created for the component. The component will not have form association, so a `ControlValueAccessor` will not be created.
+
+### Blazor integration
+
+A Blazor wrapper will be created for the component.
+
+### Visual Appearance
+
+Will use same text styling as the `nimble-table-column-text`.
```

Comment:
Despite the presence of a unit, this column still feels more like a numeric text column in spirit. One thing I could imagine a client wanting to do is have all of the values right aligned, and possibly having them all have the same amount of fractional digits.

Ultimately, if feels like we're missing some design decisions that might dictate some necessary APIs.

Reply:
I agree that fundamentally, we're displaying a numeric value, but I don't think that necessarily brings along any additional requirements compared to a text column (assuming that both are read-only).

I've added a justification for left-aligning values in the **Visual Appearance** section. That argument hinges on us not using a consistent number of decimal digits and not using a consistent unit. If we think it is worth providing those options, then right-alignment could make sense. I don't think we currently have any clients asking for that, but there's nothing keeping us from adding that support at a later time.

Let's see what the rest of the team thinks.

Reply:
I'll just note that in all of the prior art you provided, save SLE, columns displaying file size were right-aligned :). This may raise a question of whether or not we should consider that a "user expectation". Ultimately, I agree, that these are features that can be added over time. Given that, I think it might make sense to word your explanation for doing left-alignment a bit differently, in that we shouldn't say this column _won't_ do that, but that if we decide that we should allow users this flexibility we can add those features in a non-breaking way.

Reply:
I agree with @m-akinc -- the rationale we were given by visual designers was that right-aligned values made sense if all values had the same number of decimals and the same length of units.

Perhaps, this is really a question for the visual designers to see what their expectation is for this column's alignment.